src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -329,6 +326,45 @@ void AsynchronousMetrics::update()
     saveAllArenasMetric<size_t>(new_values, "muzzy_purged");
 #endif
 
+    // Try to add processor frequencies, ignoring errors.
+    try
+    {
+        ReadBufferFromFile buf("/proc/cpuinfo");
+
+        // We need the following lines:
+        // core id : 4
+        // cpu MHz : 4052.941
+        // They contain tabs and are interspersed with other info.
+        int core_id = 0;
+        while (!buf.eof())
+        {
+            std::string s;
+            readEscapedStringUntilEOL(s, buf);
```

Comment:
Who said that it can be escaped...

Reply:
It works for non-escaped lines as well, the difference is that it can parse slash escape sequences. We won't have them so probably it's OK. I'll add a comment.

Reply:
Ok.